lib/context_request_middleware/cookie.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module ContextRequestMiddleware
+  # Cookie module to consist the compatibility with
+  # rack version 1.x & 2.0
+  module Cookie
+    HTTP_COOKIE = 'HTTP_COOKIE' if Rack.release < '2.0.0'
+
+    # :nocov:
+    def session
```

Comment:
This method should be named 'session_id' right? See comment above.

Reply:
To be renamed **session_id** sounds reasonable, but if I do it.
There will be a conflict in:

> cookie_session_retriever.rb

as there is already a method called **session_id**

I suggest this:

**session** method in **lib/context_request_middleware/cookie.rb** to be renamed to:
```ruby
def cookie_session_id(request)
```

and used like this for example in **cookie_session_retriever.rb**

```ruby
def new_session_id?
    session_id && session_id != request_cookie_session_id
end

def request_cookie_session_id
    cookie_session_id(@request)
end

```

Reply:
👍